src/main/kotlin/kscript/app/creator/PackageCreator.kt

```diff
@@ -6,28 +6,51 @@ import kscript.app.model.Script
 import kscript.app.util.Executor
 import kscript.app.util.FileUtils
 import kscript.app.util.Logger.infoMsg
+import kscript.app.util.ScriptUtils.dropExtension
 import java.nio.file.Path
+import java.nio.file.Paths
+import java.io.File
 
 class PackageCreator(private val executor: Executor) {
     /**
      * Create and use a temporary gradle project to package the compiled script using capsule.
      * See https://github.com/puniverse/capsule
      */
     fun packageKscript(basePath: Path, script: Script, jarArtifact: JarArtifact): Path {
+        val baseName = script.scriptName.dropExtension()
+
         infoMsg("Packaging script '${script.scriptName}' into standalone executable...")
 
         // create exec_header to allow for direction execution (see http://www.capsule.io/user-guide/#really-executable-capsules)
         // from https://github.com/puniverse/capsule/blob/master/capsule-util/src/main/resources/capsule/execheader.sh
-        FileUtils.createFile(basePath.resolve("exec_header.sh"), Templates.executeHeader)
+        var hdr = Templates.executeHeader
+        val parts = mutableListOf("")
+
+        for (opt in script.kotlinOpts) {
+            val s = opt.value
+            if (s.startsWith("-J")) {
+                parts.add(s.substring(2))
+            }
+        }
+        val opts = parts.joinToString(" ").trim()
+        hdr = hdr.replace("java -jar", "java $opts -jar")
+        FileUtils.createFile(basePath.resolve("exec_header.sh"), hdr)
         FileUtils.createFile(
             basePath.resolve("build.gradle.kts"), GradleTemplates.createGradlePackageScript(script, jarArtifact)
         )
 
         executor.createPackage(basePath)
 
-        basePath.resolve("build/libs/appName").toFile().setExecutable(true)
+        val ef = basePath.resolve("build/libs/$baseName").toFile()
+        ef.setExecutable(true)
+
+        infoMsg("Finished packaging '${script.scriptName}'; executable path: ${basePath}/build/libs/$baseName")
```

Comment:
This message is not printed if the package is already in cache. Can we print the executable path every time, even if it comes from the cache? As we do not copy file (which I think is okay) we should at least communicate full executable path to the user.

Currently, this is printed whenever new package is created:
[kscript] Packaging script 'url_test.kts' into standalone executable...
[kscript] Finished packaging 'url_test.kts'; executable path: /home/vagrant/.kscript/cache/package_5a501be28df8e8a4ace2de5f8162d0ee/build/libs/url_test
[kscript] Package created in: /home/vagrant/.kscript/cache/package_5a501be28df8e8a4ace2de5f8162d0ee

And only this if it comes from cache:
[kscript] Package created in: /home/vagrant/.kscript/cache/package_5a501be28df8e8a4ace2de5f8162d0ee

Reply:
The current behavior is unintuitive when the package is already in cache. The exact intended behavior of `--package` is unclear - with a clean cache, the packaged executable is placed in the current directory at the end. With a populated cache, it isn't (even if it is absent) - this doesn't seem intuitive. I would expect the following behavior:
```
if (package not in cache) {
  create package in cache
  build package using Gradle
  report success or failure of build
}
if (built executable in cache) {
  copy executable to current directory
}
report package location and executable path in cache
```
Isn't this better than what we have now?

Reply:
I am not a big fan of copying files around - I would rather clearly communicate to the user where is the resulting file. Then it will be just enough to check in Cache.getOrCreatePackage() if the package file is already there or not. Currently, there is only a check for directory - if it exists then it is assumed that the package exists, which might be the wrong assumption. So definitely cache condition has to be improved: if there is a directory, but no package, PackageCreator should be started again. From my point of view, I would rather not copy files to the current directory, but in all cases (cached/not cached) communicate to user where is package located.

Reply:
An additional advantage of not copying files in 4.1 will be that PackageCreator would be testable with only Junit Tests. With OsPath there are mostly no references to the underlying system, except for some utility classes, which possibly can be mocked.